### ISSUE TITLE
Invalid cookie for this PagedSearch request error while loading next page result from ApacheDS

### ISSUE THREAD

**Issue by prashanthmadduri** [ldapjs/node-ldapjs#578]
Hi Team, 

We have ApacheDS instance and are trying to get the paged result using `PagedResultsControl` for subsequent calls with the cookie captured from previous paged result. However it is throwing `error: Invalid cookie for this PagedSearch request.` 
The sample code is below:
```
var client = ldap.createClient({
	url: 'ldap://<ldap_host>:389'
});
client.bind('<Bind DN>', '<Bind Password>', function (err) {
	if (err) {
		console.log('client.bind error', err);
	} else {
		console.log('bind is success');
		fs.readFile("./cookie1", function (err, pagedCookie) {
			if (err) {
				return console.log(err);
			}
			console.log("The file read saved!");
			var opts1 = {
				filter: '(&(objectClass=*))',
				scope: 'sub',
				attributes: [],
				attrsOnly: true
			};
			var controls = []
			controls.push(new ldap.PagedResultsControl({
				value: {
					size: 10,
					cookie: pagedCookie
				},
				criticality: true
			}))
			var count = 1;
			client.search('<Search Base>', opts1, controls, function (err, res1) {
				if (err) {
					console.log('error', err);
				} else {
					res1.on('searchEntry', function (entry) {
						console.log('page 2 entry ' + count++ + ' : ' + JSON.stringify(entry.object));
					});
					res1.on('page', function (result, cb) {
						console.log('page end');
					});
					res1.on('error', function (err) {
						console.error('error: ' + err.message);
					});
					res1.on('end', function (result) {
						console.log('result: ' + result);
						fs.writeFile("./cookie2", result.controls[0].value.cookie, function (err) {
							if (err) {
								return console.log(err);
							}
							console.log("The file was saved!");
							client.unbind(function (err) {
								if (err) {
									console.log('error unbind : ', err);
								} else {
									console.log('unbind is success');
								}
							});
						});
					});
				}
			});
		});
	}
});
```
Here we are sending cookie with `PagedResultsControl` object to get the next page:
```
controls.push(new ldap.PagedResultsControl({
				value: {
					size: 10,
					cookie: pagedCookie
				}
```
It is throwing `error: Invalid cookie for this PagedSearch request.` error and same implementation is working for Microsoft Active Directory. 

As per our finding from the RFC [https://docs.ldap.com/specs/rfc2696.txt](url) the cookie is invalid only once all the results are retrieved. 

Please advise in case anything is not correct. Thank you.

**Comment by jsumners** [comment]
If the same code works with Active Directory but not ApacheDS then I would say it is a server implementation problem, not an ldapjs library problem.

**Comment by jsumners** [comment]
I am closing this issue due to inactivity. If you continue to experience the problem, feel free to reopen.

**Comment by jsumners** [comment]
⚠️ This issue has been locked due to age. If you have encountered a recent
problem that seems to be covered by this issue, please open a new issue.

Please include a [minimal reproducible example](https://en.wikipedia.org/wiki/Minimal_reproducible_example)
when opening a new issue.
